frontend/public/components/alert.tsx

```diff
@@ -0,0 +1,427 @@
+/* eslint-disable no-undef */
+
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import { coFetchJSON } from '../co-fetch';
+import { AlertModel, AlertRuleModel } from '../models';
+import { referenceForModel } from '../module/k8s';
+import { alertRuleState } from '../module/monitoring';
+import k8sActions from '../module/k8s/k8s-actions';
+import { MonitoringRoutes, connectToURLs } from '../monitoring';
+import store from '../redux';
+import { UIActions } from '../ui/ui-actions';
+import { ColHead, List, ListHeader, ResourceRow, TextFilter } from './factory';
+import { CheckBoxes } from './row-filter';
+import { SafetyFirst } from './safety-first';
+import { history, NavTitle, ResourceIcon, resourcePath, SectionHeading, StatusBox, Timestamp, withFallback } from './utils';
+import { formatDuration } from './utils/datetime';
+
+const reduxID = 'monitoringRules';
+const alertKind = referenceForModel(AlertModel);
+const ruleKind = referenceForModel(AlertRuleModel);
+
+const detailsURL = (kind, name, labels) => `${resourcePath(kind, name)}?${_.map(labels, (v, k) => `${encodeURIComponent(k)}=${encodeURIComponent(v)}`).join('&')}`;
+
+const ExternalLink = ({href, text}) => <a className="co-external-link" href={href} target="_blank" rel="noopener noreferrer">{text}</a>;
+
+const stateToIconClassName = {
+  firing: 'fa fa-bell alert-firing',
+  pending: 'fa fa-exclamation-triangle alert-pending',
+};
+
+const State = ({state}) => {
```

Comment:
You can add `const State: React.SFC<StateProps>` and then define `StateProps` for better type safety.